File: php/easypay.app/easypay_controller.php
<?php

$endOfLine = "<br/>";
ini_set('display_errors', 'On');
error_reporting(E_ALL);

include '../easypay.lib/Direct_payment_request.php';
include '../easypay.lib/Checkout_page_request.php';
include '../easypay.lib/Easypay_config.php';
include '../easypay.lib/Easypay_request.php';
include '../easypay.lib/Message.php';
include '../easypay.lib/Signature.php';
include '../easypay.lib/Payment_response.php';
include '../easypay.lib/Status_response.php';
use easpaylib\Checkout_page_request;
use easpaylib\Easypay_config;
use easpaylib\Easypay_request;
use easpaylib\Message;

$MerchantId = "LTC001";
$BaseReturnUrl = "http://localhost/easypay.app/easypay_controller.php";

$paymentRequest = new Checkout_page_request();
$paymentRequest->set_adult_content(false);
$paymentRequest->set_amount(50);
$paymentRequest->set_payment_info("Test");
$paymentRequest->set_title("My new Phone");
$paymentRequest->set_description("Buy your favorite Sony Phone!");
$paymentRequest->set_adult_content(false);
$paymentRequest->set_roaming(false);
$paymentRequest->set_merchant_id($MerchantId);
$paymentRequest->set_cancel_url($BaseReturnUrl . "?uid=43c2&sid=dc0d&purchase=cancel");
$paymentRequest->set_error_url($BaseReturnUrl . "?uid=43c2&sid=dc0d&purchase=error");
$paymentRequest->set_success_url($BaseReturnUrl . "?uid=43c2&sid=dc0d&purchase=success");
$paymentRequest->set_cp_service_id("xys-323-gh-ff");
$paymentRequest->set_cp_subscription_id("23hkb379oh");
$paymentRequest->set_cp_user_id("vghv5678");
$paymentRequest->set_image_url("http://lorempixel.com/300/200");
$paymentRequest->set_content_type("App");
$paymentRequest->set_store_source("Easypay-City");

$config = new Easypay_config();
$config->set_basepath("/charging-engine-checkout");
$config->set_host("easypay-staging.swisscom.ch");
$config->set_merchant_id($MerchantId);
$config->set_easypay_secret("KiLgscVNTqAJQ1keGOv_hhKsuf5oftohg17VmncT");

ECHO json_encode($_GET) . $endOfLine;
ECHO "----------------------------------------------------------" . $endOfLine;

if (isset($_GET['buy'])) {

    ECHO "Buy Action" . "<br/>";
    if (isset($_GET['amount'])) {
        $paymentRequest->set_amount($_GET['amount']);
    }

    if (isset($_GET['paymentInfo'])) {
        $paymentRequest->set_payment_info($_GET['paymentInfo']);
    }

    $url = Easypay_request::get_checkout_page_url($config, $paymentRequest);
    ECHO "Redirect to " . $url . "<br/>";
    header("Location: " . $url);
    exit;

} elseif (isset($_GET['purchase'])) {
    $purchase = $_GET['purchase'];
    $paymentId = isset($_GET['paymentId']) ? $_GET['paymentId'] : null;

    if ($purchase == "success") {
        $reqId = "req-id-123";

        $config->set_basepath("/ce-rest-service");
        $response = Easypay_request::commit_payment_request($paymentId, $config, $reqId);

        if ($response->is_success())
            ECHO "<h2>Success</h2>";
        else {
            ECHO "<h2>ERROR</h2>";

            /** @var Message $msg */
            $messages = $response->get_messages();
            if (count($messages) > 0) {
                foreach ($messages as $msg) {
                    ECHO "<p><strong>" . $msg->get_code() . "</strong>, " . $msg->get_field() . ", " . $msg->get_message() . "</p>";
                }
            } else {
                ECHO "<h2>ERROR</h2>";
                ECHO "<strong>" . $response->get_exception() . "</strong>";
            }
        }
    } else {
        ECHO "<h2>ERROR</h2>";
        ECHO "<strong>Purchase terminated with state: '" . $purchase . "'</strong>";
    }

    if (isset($_GET['paymentId'])) {
        ECHO "<form method=\"get\" action=\"easypay_controller.php\"><input type=\"hidden\" name=\"paymentId\" value=\"" . $_GET['paymentId'] . "\" ><input type=\"submit\" name=\"status\" value=\"Get Payment Status\"/></form>";
    }
} elseif (isset($_GET['status']) && $_GET['paymentId']) {
    $paymentId = $_GET['paymentId'];
    $reqId = "req-id-123";

    $config->set_basepath("/ce-rest-service");
    $response = Easypay_request::get_payment_status($paymentId, $config, $reqId);

    ECHO "<h2>STATUS " . $response->get_status() . "</h2>";
    ECHO "<p><strong>MSISDN: " . $response->get_formatted_msisdn() . "</strong>, OrderID: " . $response->get_order_ID() . ", Payment Info: " . $response->get_payment_info() . ", CHF " . $response->get_amount() . "</p>";
}

File: cSharp/easypay.web/Controllers/EasypayController.cs
namespace easypay.web.Controllers
{
    using System;
    using System.Linq;
    using System.Web.Mvc;
    using System.Threading.Tasks;
    using System.Web.UI.WebControls;
    using easypay.swisscom.com;
    using easypay.web.Models;

    public class EasypayController : Controller
    {
        private const string BaseReturnUrl = "http://localhost:6392/Easypay/Callback";
        private const string MerchantId = "LTC001";

        private static EasypayConfig GetEasypayConfig()
        {
            return new EasypayConfig
            {
                Host = "easypay-staging.swisscom.ch",
                MerchantId = MerchantId,
                EasypaySecret = "KiLgscVNTqAJQ1keGOv_hhKsuf5oftohg17VmncT"
            };
        }

        // GET: Easypay
        public ActionResult Index()
        {
            return View("BuyPage");
        }

        // GET: Easypay/Callback?paymentId=xyz&purchase=success
        public async Task<ActionResult> Callback(string paymentId, string purchase)
        {
            var msg = new Response { Title = "Technical Problem", Message = "", PaymentId = paymentId};

            if (purchase == "success")
            {
                var reqId = msg.ReqId = Guid.NewGuid().ToString();

                var easypayConfig = GetEasypayConfig();
                easypayConfig.Basepath = "/ce-rest-service";

                try
                {
                    var message = await EasypayRequest.CommitPaymentRequest(paymentId, easypayConfig, reqId);

                    if (message.Success)
                        msg.Title = "Success";
                    else
                        msg.Message = message.Messages.Select(x => string.Join(", ", x.Code, x.Field, x.RequestId)).FirstOrDefault();
                }
                catch (Exception ex)
                {
                    msg.Message = ex.Message;

                }
            }
            else
            {
                msg.Title = "Purchase terminated with state: '" + purchase + "'";
            }

            return View("ResponsePage", msg);
        }

        // GET: Easypay
        public async Task<ActionResult> Status(string paymentId, string reqId)
        {
            var easypayConfig = GetEasypayConfig();
            easypayConfig.Basepath = "/ce-rest-service";

            var message = await EasypayRequest.GetPaymentStatus(paymentId, easypayConfig, reqId);
            var msg = new Response { Title = "STATUS: " + message.Status, Message = "Msisdn: " + message.FormattedMsisdn + ", OrderID: " + message.OrderID + ", Payment Info: " + message.PaymentInfo + ", CHF " + message.Amount, PaymentId = paymentId, ReqId = reqId };

            return View("ResponsePage", msg);
        }

        [HttpPost]
        public ActionResult Buy()
        {
            var paymentRequest = new CheckoutPageRequest();
            paymentRequest.AdultContent = false;
            paymentRequest.Amount = "5.50";
            paymentRequest.PaymentInfo = "Test";
            paymentRequest.Title = "My new Phone";
            paymentRequest.Description = "Buy your favorite Sony Phone!";
            paymentRequest.AdultContent = false;
            paymentRequest.Roaming = false;
            paymentRequest.MerchantId = MerchantId;
            paymentRequest.CancelUrl = BaseReturnUrl + "?uid=43c2&sid=dc0d&purchase=cancel";
            paymentRequest.ErrorUrl = BaseReturnUrl + "?uid=43c2&sid=dc0d&purchase=error";
            paymentRequest.SuccessUrl = BaseReturnUrl + "?uid=43c2&sid=dc0d&purchase=success";
            paymentRequest.CpServiceId = "xys-323-gh-ff";
            paymentRequest.CpSubscriptionId = "23hkb379oh";
            paymentRequest.CpUserId = "vghv5678";
            paymentRequest.ImageUrl = "http://lorempixel.com/300/200";
            paymentRequest.ContentType = "App";
            paymentRequest.StoreSource = "Easypay-City";

            var easypayConfig = GetEasypayConfig();
            easypayConfig.Basepath = "/charging-engine-checkout";
            var url = EasypayRequest.GetCheckoutPageUrl(easypayConfig, paymentRequest);
            return Redirect(url.ToString());
        }
    }
}